data_steward/retraction/retract_data_bq.py

```diff
@@ -417,28 +423,44 @@ def skip_dataset_retraction(dataset_id, retraction_type) -> bool:
     return False
 
 
-def get_primary_key_for_sandbox_table(table: str) -> str:
+def get_primary_key_for_sandbox_table(client: BigQueryClient, dataset,
+                                      table: str) -> str:
     """
     NOTE This is intended for `only_ehr` retraction for sandbox datasets.
-    
+
     When retracting sandbox tables for `only_ehr`, map/ext tables do not
     exist for them. We must look at its primary key instead, in addition to person_id.
-    
-    This function looks at the sandbox table name, gets its domain, and returns the key
-    we must use for retraction. 
 
+    This function looks at the sandbox table name, gets its domain, checks if 
+    the domain_id and person_id columns exists in the table, and returns the domain_id
+    we can use for retraction. 
+
+    :param client: BigQuery client
+    :param dataset: Dataset to run retraction on
     :param table: name of the sandbox table that needs retraction
     :return: column name that the retraction must run on. 
         Returns '' if the domain table cannot be identified, the domain table
-        does not contain person_id, or the domain table does not contain EHR data.
+        does not contain person_id and/or domain_id, or the domain table does 
+        not contain EHR data.
     """
-    domain_tables = set(AOU_REQUIRED) - set(NON_PID_TABLES)
+    domain_tables = set(AOU_REQUIRED + OTHER_PID_TABLES) - set(NON_PID_TABLES)
 
     for domain in domain_tables:
-        if domain in table:
+        if table.endswith(domain):
+
             if domain == DEATH:
-                return f"{PERSON}_id"
-            return f"{domain}_id"
+                domain_id = f"{PERSON}_id"
+
+            elif domain == OBSERVATION and OBSERVATION_PERIOD in table:
+                domain_id = f"{OBSERVATION_PERIOD}_id"
```

Comment:
Since we are using .endswith, this situation may not arise. Also, there must be some other place in the codebase (maybe in unioned) where there is an hpo prefix added to omop table names and we need to determine the omop table. You can try to replicate that.